### PR TITLE
Feature/main camera marker

### PR DESCRIPTION
**Pull request by dabeshouse**
Set the camera marker buttons to always affect the main camera.

### CODE REVIEW

File: core/ide/src/main/java/org/alice/stageide/sceneeditor/StorytellingSceneEditor.java [hunk 1]
```diff
@@ -1022,6 +1015,8 @@ protected void setActiveScene(UserField sceneField) {
 
       MoveActiveCameraToMarkerActionOperation.getInstance().setCamera(movableSceneCameraImp);
```
Comment (jennej) [inline core/ide/src/main/java/org/alice/stageide/sceneeditor/StorytellingSceneEditor.java:816]: should these be startingCameraMarkerImp too?

File: core/story-api/src/main/java/org/alice/interact/DragAdapter.java [hunk 2]
```diff
@@ -954,8 +944,11 @@ public enum CameraView {
   }
```
Comment (jennej) [inline core/story-api/src/main/java/org/alice/interact/DragAdapter.java:945]: CameraPair might not be the right name for this classlet

File: core/ide/src/main/java/org/alice/stageide/sceneeditor/viewmanager/MarkerUtilities.java [hunk 3]
```diff
@@ -412,33 +357,6 @@ public static Color getColorForMarkerField(UserField markerField) {
...
-
-  private static String getNameForColor(Color color) {
-    int colorIndex = getColorIndexForColor(color);
-    if (colorIndex != -1) {
-      return getColorNameForIndex(colorIndex);
-    } else {
-      return "";
-    }
-  }
-
```
Comment (jennej) [inline core/ide/src/main/java/org/alice/stageide/sceneeditor/viewmanager/MarkerUtilities.java:441]: nice cleanup on this file.  we'll get icons sane eventually

File: core/ide/src/main/java/org/alice/stageide/sceneeditor/viewmanager/CameraMarkerTracker.java [hunk 4]
```diff
@@ -115,139 +110,543 @@ public void setCameras(SymmetricPerspectiveCamera perspectiveCamera, Orthographi
...
+    }
+
+    @Override
+    protected void centerOn(AxisAlignedBox box) {
+      double targetHeight = box.getHeight();
+      double targetWidth = box.getWidth();
+      double targetDepth = box.getDepth();
+      Point3 targetTranslation = box.getCenter();
+
+      // Update Orthographic camera markers
```
Comment (jennej) [inline core/ide/src/main/java/org/alice/stageide/sceneeditor/viewmanager/CameraMarkerTracker.java]: can remove the  // Update ... comment,  it was just to break up that massive function, but you've done that even better.

File: core/ide/src/main/java/org/alice/stageide/sceneeditor/StorytellingSceneEditor.java [hunk 5]
```diff
@@ -604,112 +488,30 @@ private boolean isSelectableType(AbstractType<?, ?, ?> valueType) {
...
-    this.snapGrid.addCamera(perspectiveCamera);
+  private void setCameras() {
+    SymmetricPerspectiveCamera mainCamera = sceneCameraImp.getSgCamera();
+    SymmetricPerspectiveCamera layoutCamera = layoutCameraImp.getSgCamera();
+    OrthographicCamera orthographicCamera = orthographicCameraImp.getSgCamera();
+    this.globalDragAdapter.addCameraView(CameraView.MAIN, mainCamera, layoutCamera, orthographicCamera);
+    this.globalDragAdapter.makeCameraActive(mainCamera);
+    this.mainCameraViewTracker.setCameras(mainCamera, layoutCamera, orthographicCamera);
+    this.snapGrid.addCamera(mainCamera);
+    snapGrid.addCamera(layoutCamera);
```
Comment (jennej) [inline core/ide/src/main/java/org/alice/stageide/sceneeditor/StorytellingSceneEditor.java:512]: you have a single snapGrid without a this here.  we should at least be consistent at the function level 😅

File: core/ide/src/main/java/org/alice/stageide/sceneeditor/viewmanager/CameraMarkerTracker.java [hunk 6]
```diff
@@ -115,139 +110,543 @@ public void setCameras(SymmetricPerspectiveCamera perspectiveCamera, Orthographi
...
+    protected abstract void initialize();
+    protected abstract void centerOn(AxisAlignedBox box);
+
+    public abstract void resetForScene(SceneImp sceneImp, AffineMatrix4x4 startingView);
+
+    protected abstract void stopTrackingCamera();
+
+    // We're not changing the active/selected marker, but we may have moved it
+    protected void updateCameraToNewMarkerLocation() {
+      // because the local transform of the marker has been set outside this class, we need to temporarily unhook the
```
Comment (jennej) [inline core/ide/src/main/java/org/alice/stageide/sceneeditor/viewmanager/CameraMarkerTracker.java]: the first part comment was relevant when we were setting the positions in StorytellingSceneEditor, but it isn't true now.

**Review by jennej — APPROVED** [review]
just a few nitpicky bits.  

the DragAdaptor class gives me feelings, but nothing that needs to get pulled into these changes

File: core/story-api/src/main/java/org/alice/interact/DragAdapter.java (hunk 2 above)
Comment (dabeshouse) [reply core/story-api/src/main/java/org/alice/interact/DragAdapter.java:945]: Oh yeah, I meant to change that.

File: core/ide/src/main/java/org/alice/stageide/sceneeditor/viewmanager/MarkerUtilities.java (hunk 3 above)
Comment (dabeshouse) [reply core/ide/src/main/java/org/alice/stageide/sceneeditor/viewmanager/MarkerUtilities.java:441]: Eventually, but not in this PR :(

File: core/ide/src/main/java/org/alice/stageide/sceneeditor/StorytellingSceneEditor.java (hunk 5 above)
Comment (dabeshouse) [reply core/ide/src/main/java/org/alice/stageide/sceneeditor/StorytellingSceneEditor.java:512]: I cannot explain this, but I can correct it. I am not found of the extra `this`s so I am cleaning this^H^H^H^H method, and stopping myself from doing so at a more global level.

File: core/ide/src/main/java/org/alice/stageide/sceneeditor/viewmanager/CameraMarkerTracker.java (hunk 6 above)
Comment (dabeshouse) [reply core/ide/src/main/java/org/alice/stageide/sceneeditor/viewmanager/CameraMarkerTracker.java]: I changed the comments. Hopefully they are correct now.

**Comment by dabeshouse** [comment]
> the DragAdaptor class gives me feelings, but nothing that needs to get pulled into these changes

Feelings? Like mountain climbing and suddenly finding a cliff dropping off right in front of you?

**Pull request merged**